[dtensor] remove custom dispatch op

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97583
* __->__ #95629

Since we removed all custom dispatch ops, we can safely delete this
table as we won't use it for other purposes

wanchaol committed:
[dtensor] remove custom dispatch op

Since we removed all custom dispatch ops, we can safely delete this
table as we won't use it for other purposes

[ghstack-poisoned]

XilunWu reviewed (approved):
Great to see `register_impl` can be removed. nit: fix linter

wanchaol committed:
Update on "[dtensor] remove custom dispatch op"


Since we removed all custom dispatch ops, we can safely delete this
table as we won't use it for other purposes

[ghstack-poisoned]

wanchaol committed:
Update on "[dtensor] remove custom dispatch op"


Since we removed all custom dispatch ops, we can safely delete this
table as we won't use it for other purposes

[ghstack-poisoned]

wanchaol commented:
@wanchaol has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44384419).

wanchaol committed:
Update on "[dtensor] remove custom dispatch op"


Since we removed all custom dispatch ops, we can safely delete this
table as we won't use it for other purposes

Differential Revision: [D44384419](https://our.internmc.facebook.com/intern/diff/D44384419)

[ghstack-poisoned]
